addons/controllers/clusterbootstrap_controller.go

```diff
@@ -575,6 +574,21 @@ func (r *ClusterBootstrapReconciler) mergeClusterBootstrapPackagesWithTemplate(
 			updatedClusterBootstrap.Spec.AdditionalPackages = append(updatedClusterBootstrap.Spec.AdditionalPackages, newPkg)
 			packages = append(packages, newPkg)
 		}
+		additionalPackageTemplateMap[packageRefName] = templatePkg
+	}
+
+	// There might be packages in clusterbootstrap but not in clusterboostraptemplate.
+	// The below loop tries to find those packages and updates the RefName if it is found in tkr's spec.bootstrapPackages.
+	// If its not there in tkr's spec.bootstrappackages then user has to update the version of it in clusterbootstrap.
+	for additionalPkgRefName, additionalPkg := range additionalPackageMap {
+		if _, ok := additionalPackageTemplateMap[additionalPkgRefName]; !ok {
+			additionalPkgFromTKr, additionalPkgNamePrefix, err := util.GetBootstrapPackageNameFromTKR(r.context, r.Client, additionalPkg.RefName, cluster)
+			if err != nil {
+				log.Info(fmt.Sprintf("unable to find any additional bootstrap package prefixed with '%s' for ClusterBootstrap %s/%s in TKR", additionalPkgNamePrefix, cluster.Name, cluster.Namespace))
```

Comment:
Seems like when error is returned on ```GetBootstrapPackageNameFromTKR```, it does not return prefix information any more. Not related to this PR but a bug with existing logic. Could you please modify the ```additionalPkgNamePrefix``` to return prefix even on error so that the log above can be useful?

Reply:
Also I noticed in most places we log with this format & ordering ```<NAMESPACE>/<NAME>```. Could you please fix this in the log message above and in the other existing usage of ```GetBootstrapPackageNameFromTKR``` in the code to stay consistent?

Reply:
Fixed